File: EmptyProject/WorkflowsTrainingTwo/WorkflowsTraining/Parsing/Expressions/Nodes/Expressions/StringLiteralNode.cs
using WorkflowsTraining.Parsing.Expressions.Interfaces;

namespace WorkflowsTraining.Parsing.Expressions.Nodes.Expressions;

public class StringLiteralNode(string value) : IExpressionNode
{
    private string Value { get; } = value;

    public object Resolve()
    {
        return Value;
    }
}

File: EmptyProject/WorkflowsTrainingTwo/WorkflowsTrainingCPlus/Parsing/Expressions/Nodes/Expressions/StringLiteralNode.cpp
#include "StringLiteralNode.h"
#include <iostream>

namespace WorkflowsTraining::Parsing::Expressions::Nodes::Expressions {
    StringLiteralNode::StringLiteralNode(std::string value)
        : value_(std::move(value)) {
    }

    std::variant<bool, int, double, std::string> StringLiteralNode::Resolve() {
        std::cout << "StringLiteralNode: Resolving to \"" << value_ << "\"" << std::endl;
        return value_;
    }
}